subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheClassLoaderScopeRegistryListener.kt

```diff
@@ -58,41 +63,34 @@ class ConfigurationCacheClassLoaderScopeRegistryListener : ClassLoaderScopeRegis
         //  The listener only needs to be attached in the `store` state.
         scopeSpecs.clear()
         loaders.clear()
-        detach()
+        listenerManager.remove(this)
     }
 
     override fun close() {
         dispose()
     }
 
-    private
-    fun detach() {
-        manager?.removeListener(this)
-        manager = null
-    }
-
     override fun scopeFor(classLoader: ClassLoader?): Pair<ClassLoaderScopeSpec, ClassLoaderRole>? {
         return loaders[classLoader]
     }
 
-    override fun rootScopeCreated(rootScopeId: ClassLoaderScopeId) {
```

Comment:
There was no real reason to serialize these scopes, as they are static across builds.